File: .tmp_py4013/batch_13_dump/3445_maximum_difference_between_even_and_odd_frequency_ii.js
// LeetCode 3445 - Maximum Difference Between Even and Odd Frequency II
// https://leetcode.com/problems/maximum-difference-between-even-and-odd-frequency-ii/

var maxDifference = function(s, k) {
    const n = s.length;
    let ans = -1e9;
    for (let a = 0; a < 5; a++) {
        for (let b = 0; b < 5; b++) {
            if (a === b) continue;
            const prefA = new Array(n + 1).fill(0), prefB = new Array(n + 1).fill(0);
            for (let i = 0; i < n; i++) {
                prefA[i + 1] = prefA[i];
                prefB[i + 1] = prefB[i];
                if (s.charCodeAt(i) - 48 === a) prefA[i + 1]++;
                if (s.charCodeAt(i) - 48 === b) prefB[i + 1]++;
            }
            for (let i = 0; i < n; i++) {
                for (let j = i + k - 1; j < n; j++) {
                    const fa = prefA[j + 1] - prefA[i];
                    const fb = prefB[j + 1] - prefB[i];
                    if (fa % 2 === 1 && fb % 2 === 0 && fb > 0) {
                        if (fa - fb > ans) ans = fa - fb;
                    }
                }
            }
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check17/3445_maximum_difference_between_even_and_odd_frequency_ii.rs
struct Solution;
// LeetCode 3445 - Maximum Difference Between Even and Odd Frequency II
// https://leetcode.com/problems/maximum-difference-between-even-and-odd-frequency-ii/

impl Solution {
    pub fn max_difference(s: String, k: i32) -> i32 {
        let n = s.len();
        let bytes = s.as_bytes();
        let mut ans = -1_000_000_000;
        for a in 0..5 {
            for b in 0..5 {
                if a == b {
                    continue;
                }
                let mut pref_a = vec![0; n + 1];
                let mut pref_b = vec![0; n + 1];
                for i in 0..n {
                    pref_a[i + 1] = pref_a[i];
                    pref_b[i + 1] = pref_b[i];
                    if (bytes[i] - b'0') as i32 == a {
                        pref_a[i + 1] += 1;
                    }
                    if (bytes[i] - b'0') as i32 == b {
                        pref_b[i + 1] += 1;
                    }
                }
                for i in 0..n {
                    let mut j = i + k as usize - 1;
                    while j < n {
                        let fa = pref_a[j + 1] - pref_a[i];
                        let fb = pref_b[j + 1] - pref_b[i];
                        if fa % 2 == 1 && fb % 2 == 0 && fb > 0 && fa - fb > ans {
                            ans = fa - fb;
                        }
                        j += 1;
                    }
                }
            }
        }
        ans
    }
}

fn main() {}
